### PR TITLE
add MultiPath TCP support in kernel version >= 5.6

### PR DESCRIPTION
**Pull request by sKyissKy**
Thanks for your contribution to OpenWrt!

To help keep the codebase consistent and readable,
and to help people review your contribution,
we ask you to follow the rules you find in the wiki at this link
https://openwrt.org/submitting-patches

Please remove this message before posting the pull request.

### CODE REVIEW

File: target/linux/generic/config-5.10 [hunk 1]
```diff
@@ -3523,7 +3523,9 @@ CONFIG_MODULE_UNLOAD=y
 # CONFIG_MPLS is not set
 # CONFIG_MPLS_IPTUNNEL is not set
 # CONFIG_MPLS_ROUTING is not set
-# CONFIG_MPTCP is not set
+CONFIG_MPTCP=y
+CONFIG_INET_MPTCP_DIAG=m
```
Comment (hauke) [inline target/linux/generic/config-5.10:3527]: building modesl here is useless. If you need the module please create a kmod package in package/kernel/linux/modules/

**Comment by hauke** [comment]
What is the size increase of a kernel image with this change?

File: target/linux/generic/config-5.10 (hunk 1 above)
Comment (hauke) [reply target/linux/generic/config-5.10]: The `CONFIG_INET_MPTCP_DIAG` option should better go to `kmod-inet-diag` in `package/kernel/linux/modules/netsupport.mk` Then you can also package the .ko file there.

Comment (sKyissKy) [reply target/linux/generic/config-5.10]: Not all kernels (like 5.4) support multipath TCP protocol，Modifying netsupport.mk needs to add the kernel version limit, but I don't know how to add it

Comment (john-tho) [reply target/linux/generic/config-5.10]: > Not all kernels (like 5.4) support

Does not matter here (on master branch): Only 5.10 and 5.15 kernels, but, for example, if you did not want your package to build on 5.10, you could use: `DEPEND+=@!LINUX_5.10`. `FILES`, `KCONFIG`, and `AUTOLOAD` support `setting@ge5.10`: with the `@` and following, setting would only apply greater or equal to kernel 5.10. also available: le, lt, gt

> I don't know how to add it

Have a look at KernelPackage/fs-cifs for a real example of kernel version conditionals in a kernel module package: https://github.com/openwrt/openwrt/blob/96aa052c406d460de8c4c1c3cf0eb9a900ae09c5/package/kernel/linux/modules/fs.mk#L86

Also, please do not open new PRs for the same topic: https://github.com/openwrt/openwrt/pull/11071. Normally if you want a new feature in older branches, it needs to get into master first.

**Comment by abajk** [comment]
I think multipath TCP doesn't make sense in a typical router environment.

**Comment by john-tho** [comment]
> I think multipath TCP doesn't make sense in a typical router environment.

Maybe not by default, but it could go with the other conditional kernel options in [config/Config-kernel.in](https://github.com/openwrt/openwrt/blob/master/config/Config-kernel.in)

@sKyissKy please let us know if you are going to abandon this?

**Comment by sKyissKy** [comment]
After reading your comments, I am trying to write a kernel patch KO to
provide the option of multi-path TCP.

john-tho ***@***.***> 于2022年10月24日周一 06:32写道：

**Pull request closed without merging**